### DISCUSSION TITLE
remove MultitenancyStrategy

### DISCUSSION THREAD

**Discussion by gavinking** [hibernate/hibernate-orm#3866] [Design Proposals]
The `MultitenancyStrategy` enum is currently not useful.

- `MultitenancyStrategy.DISCRIMINATOR` is a nice idea, but it's never been implemented 
- There's actually no difference at all between `MultitenancyStrategy.SCHEMA` and `MultitenancyStrategy.DATABASE`, and it's misleading to have them represented by two different enumerated values.

I believe this thing could be removed with almost no impact on users at all.

**Comment by yrodiere** [comment]
On a side note, Hibernate Search needs to determine whether multi-tenancy is enabled or not. And it needs to do so very early, basically just after `Metadata` creation. Currently it does so by inspecting the `MultitenancyStrategy`. We don't care about the exact strategy, though.

So to anyone doing this change: please leave some way to detect that :)

See https://github.com/hibernate/hibernate-search/blob/21c2f0507f761bb80d74738870bcac7f4a741c60/mapper/orm/src/main/java/org/hibernate/search/mapper/orm/mapping/impl/HibernateOrmMappingInitiator.java#L74-L79

**Reply by gavinking** [reply]
Yeah, sure, we're still going to need at least *some* setting to turn it on, even if it turns out to just be the `MULTI_TENANT_CONNECTION_PROVIDER`. And that information's still going to be available via the `ConfigurationService`.

**Comment by sebersole** [comment]
> * `MultitenancyStrategy.DISCRIMINATOR` is a nice idea, but it's never been implemented

But the hope is to do it someday.  I don't have a strong opinion about this atm, but it would suck to remove it and add it back later
